### PR TITLE
Update Manage resources section

### PR DESCRIPTION
**Pull request by AnouckColson**
## Why

It is now possible to set a resource initialization strategy via the CLI or Console when branching, merging, activating an environment, and restoring a backup.

It is also possible to run a CLI command to sync resources (on top of code and/or data) when syncing 2 environments.

## What's changed

Full rework of the page to reorganize content and add sections where needed.

Closes https://github.com/platformsh/platformsh-docs/issues/3825

### CODE REVIEW

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [inline sites/friday/src/manage-resources/resource-init.md:282]: @nicogommen Is that correct?

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [inline sites/friday/src/manage-resources/resource-init.md:308]: @nicogommen Same question?

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [inline sites/friday/src/manage-resources/resource-init.md:360]: @nicogommen I'm unsure if I should talk about Production vs. Staging, or use more generic temrinology => source environment vs. target environment. WDYT?

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [inline sites/friday/src/manage-resources/resource-init.md:348]: @nicogommen Unsure about the actual step by step. Needs testing.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:52]: For the manual strategy, I think it would be nice to add also something explaining why this strategy could be a good option for the user, depending on their use case.
Because with the current description, with the failure etc., it doesn't sound very appealing 😅
The benefit of this strategy is that you can directly deploy with the desired resources. Whereas with the other strategies, if it's not exactly the resources that you want/need, you need to do a second deployment to change them. Which means that your env will run for a short moment with not exactly the wished resources. And the 2 deployments both involve some downtime (until we have real zero downtime deployments).

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:211]: To avoid confusion, I think we should add the `upsun` command at the beginning.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:217]: To avoid confusion, I think we should add the `upsun` command at the beginning.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:257]: To avoid confusion, I think we should add the `upsun` command at the beginning.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:263]: To avoid confusion, I think we should add the `upsun` command at the beginning.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [reply sites/friday/src/manage-resources/resource-init.md:282]: Yes, that's correct.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:295]: To avoid confusion, I think we should add the `upsun` command at the beginning.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:301]: To avoid confusion, I think we should add the `upsun` command at the beginning.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:327]: To avoid confusion, I think we should add the `upsun` command at the beginning.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:333]: To avoid confusion, I think we should add the `upsun` command at the beginning.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:379]: Here it gives `Upsun  `, with an uppercase U and 2 spaces after.
It should be `upsun`.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:198]: We don't need to mention this prerequisite to the user.
It's more something internal that the user doesn't need to know.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:246]: We don't need to mention this prerequisite to the user.
It's more something internal that the user doesn't need to know.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:290]: We don't need to mention this prerequisite to the user.
It's more something internal that the user doesn't need to know.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:316]: We don't need to mention this prerequisite to the user.
It's more something internal that the user doesn't need to know.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:368]: We don't need to mention this prerequisite to the user.
It's more something internal that the user doesn't need to know.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:238]: It's actually not exactly like that.
The resource init strategy is only used for new apps/services/workers.

For instance if you have a php app + a mariadb service, and you would like to add redis.
You will create a new branch, add redis, and once ready, merge this branch to the parent to deploy the new service to the parent.
It's only the new redis service that will be initialized with the chosen strategy. The php app and the mariadb service won't be touched, because we don't want to change the resources of these already running service, whereas redis is a new service that is being created.

We are actually working on a change in the Console, to show the resources init option only if we detect that a new container will be deployed with the merge.
So the modal won't show the option and stay very simple (as it is today in production), and the option will only be displayed if it is relevant, because we detected that a new container will be created with the merge.

Let me know if you have questions 🙂

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [reply sites/friday/src/manage-resources/resource-init.md:308]: At the moment, it is correct, yes.
But, we are maybe changing this... but it's not sure yet that it will make it for the release... 😅  
The change would be that we are saving the resources configured at the moment the backup is created. So when a backup is restored, we restore the environment with the exact resources as when the backup was taken.
This would be like a new strategy ("backup resources", or something like that), and it would be the default behaviour when restoring a backup.
Then you can still choose another strategy, as you described it.
So maybe we can put this section on hold for a short time. We should know very soon if this will make it in Git 26 or not.
I will let you know. 
Sorry about this change.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [reply sites/friday/src/manage-resources/resource-init.md:348]: I think this is correct 👍

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [reply sites/friday/src/manage-resources/resource-init.md:360]: Yes, I think a generic terminology would be more appropriate here.
We could use parent and child (it synchronizes from the parent to the child env).

And it is actually not exactly working like described here.

You can synchronize only the code, only the data, or only the resources, or any combination or these.
The resources sync will actually synchronize all the resources from the parent to the child (CPU&RAM, instances and disk).

Then, if you sync the data, even if you didn't choose to sync the resources, we will do what we described before:

> In this case, to ensure that the synchronization succeeds, Upsun automatically adjusts the target environment’s disk size so it matches the source environment’s disk size.

Let me know if it's not clear 🙂

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md]: @nicogommen Do you think this change in the Console might be ready for GA?

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [inline sites/friday/src/manage-resources/resource-init.md:189]: Something new with the branch operation => even when using the another strategy than parent (such as `default` or `minimum`, since we removed `manual`), we are initializing the disk size with the same disk as the parent.
This is for the same reasons as when syncing the data from parent to the child. The child needs to have enough disk to clone the data from the parent. So even if the user choose the default or minimum strategy, the disk size will be the one from the parent.

File: sites/friday/src/manage-resources/resource-init.md
Comment (tejapsh) [reply sites/friday/src/manage-resources/resource-init.md:51]: We don't support `manual` in case of backup restore either. It suffers from the same problem as having manual for branch.

File: sites/friday/src/manage-resources/resource-init.md
Comment (tejapsh) [reply sites/friday/src/manage-resources/resource-init.md:269]: I am not sure if we are implying this elsewhere or if we even need to say it, but, if the deployment generated from a `resources:set` fails, in some cases, the resources configuration will not be saved and therefore, the customer will have to set the resources again.

This usually happens if either the yaml validation or the deployment validation fails.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [reply sites/friday/src/manage-resources/resource-init.md:51]: Manual has now been removed from env activate too.
I updated my comment above accordingly.

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md:199]: Note added at the end of the section.

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md:300]: Note added at step 3.

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md]: Backup restoration section updated.

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md]: Updated to apply your suggestions.

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md]: Added a table at the end of the **Specify a resource initialization strategy** section + an info block at the beginning of every sub-section in case users go straight to one of them without reading the intro.

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md:388]: **Backup and restoration** section updated.

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md]: Done!

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md]: Updated the section with more examples of commands to give a better idea of what can be done/how.

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md]: Done!

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md:308]: I've added notes about this to the **Resource configuration** page.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [reply sites/friday/src/manage-resources/resource-init.md:199]: Thanks, the note is great.
We could maybe just put it below the tabs, because at the moment, it only appears in the `Using the CLI tab`, and not in the `In the Console` tab.

Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md:199]: You're absolutely right, I've moved the note up.

File: sites/friday/src/manage-resources/resource-init.md
Comment (nicogommen) [reply sites/friday/src/manage-resources/resource-init.md:270]: Yes, that's correct.
The resources init strategy is only applied to initialize new containers. Apps/services already running on the target env are left untouched, in terms of resources.

File: sites/friday/src/manage-resources/resource-init.md
Comment (gilzow) [inline sites/friday/src/manage-resources/resource-init.md:50]: There is no anchor on this page with an id of `default-resources` so Hugo is interpreting this as an empty `href`

**Review by gilzow — CHANGES_REQUESTED** [review]

File: sites/friday/src/manage-resources/resource-init.md
Comment (AnouckColson) [reply sites/friday/src/manage-resources/resource-init.md:54]: We decided to remove such information from this table to make it a bit lighter/easier to read, and introduced a second table a bit further down to show exactly which strategies are available when + info notes in every action section, in case readers jump straight to them without reading the intro.

**Review by Mupsi — APPROVED** [review]

**Pull request merged**
